### PR TITLE
Update go-tdx-guest version to v0.3.1

### PR DESCRIPTION
**Pull request by vbalain**
Update go-tdx-guest version to v0.3.1 which includes -
1. go-configfs-tsm moved from go-tdx-guest to its separately managed git repo.
2. Logging added to client package for QuoteProvider/Device.
3. Updated TCB Info verification logic.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/google/go-attestation v0.5.0
 	github.com/google/go-cmp v0.5.9
 	github.com/google/go-sev-guest v0.9.3
-	github.com/google/go-tdx-guest v0.2.3-0.20231222042644-aeefcb0d0eb3
+	github.com/google/go-tdx-guest v0.3.1-0.20240216171427-43efd1ff6ae8
```
Comment (alexmwu) [inline go.mod:9]: why is this not just v0.3.1?

Comment (vbalain) [reply go.mod]: I had used 'go get ...' command with main instead of latest. Fixed!

**Review by jkl73 — APPROVED** [review]

**Comment by jkl73** [comment]
The flaky failing test should be fixed by #415 , I'll merge the change once the current build is finished

**Pull request merged**